### PR TITLE
Cleanup library code

### PR DESCRIPTION
**Pull request by donny-dont**
General cleanups like

* Use `=>` when possible
* Use `const` constructors when possible
* Remove unused imports
* Use method cascades
* Use `var`

Also makes `_checkResponseSuccess` use `getUrl` and check that the response text is not empty rather than null as `reasonPhrase` is never `null`

### CODE REVIEW

File: lib/src/base_client.dart [hunk 1]
```diff
@@ -55,16 +56,15 @@ abstract class BaseClient implements Client {
 
   Future<Response> send(Request request);
 
+  void close() {}
+
   /// Throws an error if [response] is not successful.
-  void _checkResponseSuccess(url, Response response) {
+  static void _checkResponseSuccess(url, Response response) {
```
Comment (nex3) [inline lib/src/base_client.dart]: I don't generally like making helper methods static unless there's a good reason to do so.

File: lib/src/base_client.dart [hunk 2]
```diff
@@ -55,16 +56,15 @@ abstract class BaseClient implements Client {
...
     if (response.statusCode < 400) return;
-    var message = "Request to $url failed with status ${response.statusCode}";
-    if (response.reasonPhrase != null) {
-      message = "$message: ${response.reasonPhrase}";
+    var message = 'Request to $url failed with status ${response.statusCode}';
```
Comment (nex3) [inline lib/src/base_client.dart]: Please don't change the quotes. The change isn't worth the churn, and trying to keep quotes consistent isn't worth the mental overhead. Worth noting that it's explicitly *not* part of the style guide.

File: lib/src/middleware.dart [hunk 3]
```diff
@@ -48,17 +48,15 @@ Middleware createMiddleware(
...
+  return (inner) => new HandlerClient(
+        (request) => requestHandler(request)
+            .then((req) => inner.send(req))
+            .then((res) => responseHandler(res), onError: errorHandler),
+        onClose == null
+            ? inner.close
+            : () {
+                onClose();
+                inner.close();
+              },
```
Comment (nex3) [inline lib/src/middleware.dart]: No trailing commas.

File: lib/src/middleware.dart [hunk 4]
```diff
@@ -48,17 +48,15 @@ Middleware createMiddleware(
...
-          .then((res) => responseHandler(res), onError: errorHandler),
-      onClose == null
-          ? inner.close
-          : () {
-              onClose();
-              inner.close();
-            },
-    );
-  };
+  return (inner) => new HandlerClient(
```
Comment (nex3) [inline lib/src/middleware.dart]: I'm not convinced this `=>` change actually makes this more readable. There's already a *lot* of nesting and a lot of anonymous functions in the `HandlerClient` initialization, so I think having it clearly be a statement of its own makes this read more clearly.

File: test/utils.dart [hunk 5]
```diff
@@ -107,7 +103,7 @@ class _MultipartBodyMatches extends Matcher {
 ///
 /// [message] can be a String or a [Matcher].
 Matcher isClientException([message]) => predicate((error) {
-      expect(error, new isInstanceOf<http.ClientException>());
+      expect(error, const isInstanceOf<http.ClientException>());
```
Comment (nex3) [inline test/utils.dart]: We generally use `new isInstanceOf`, might as well keep this consistent.

Comment (donny-dont) [reply test/utils.dart]: Any particular rules for when you want `const` used?

File: lib/src/base_client.dart (hunk 1 above)
Comment (donny-dont) [reply lib/src/base_client.dart]: Using `static` is a bit more internally consistent with the rest of the code.

Comment (nex3) [reply lib/src/base_client.dart]: How so? The only other places I see `static` are where it's necessary to use the methods from constructors.

File: test/utils.dart (hunk 5 above)
Comment (nex3) [reply test/utils.dart]: I usually avoid it in general unless it's in a place where allocations may affect performance.

File: lib/src/base_client.dart (hunk 1 above)
Comment (donny-dont) [reply lib/src/base_client.dart]: Sorry you're correct I missed where the other ones were being called.

**Comment by donny-dont** [comment]
@nex3 think this is good to go

**Pull request merged**